File: problems/競プロ典型90/076.py
N = int(input())
A = list(map(int,input().split()))
ans = sum(A)
B = A[:]
L = A + B
cnt = 0

r = 0
l = 0
tmp = 0
while r < len(L):
    if tmp < N:
        cnt += L[r]
        tmp += 1
    else:
        tmp -= 1
        cnt -= L[l]
        l += 1
    
    if cnt * 10 == ans:
        print("Yes")
        exit()
    elif cnt * 10 > ans:
        while cnt * 10 > ans:
            tmp -= 1
            cnt -= L[l]
            l += 1
    r += 1

print("No")

File: problems/競プロ典型90/076.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;

#define rep(i, a, b) for(int i = a; i < b; i++)
#define rrep(i, a, b) for(int i = a; i >= b; i--)
#define fore(i, a) for(auto &i : a)

#define all(x) (x).begin(),(x).end()

int main(){
    int N;
    cin >> N;
    vector<int> A(N);
    ll sum = 0;
    rep(i, 0, N) {
        cin >> A.at(i);
        sum += A.at(i);
    }
    vector<int> v(2*N);
    rep(i, 0, N) v.at(i) = A.at(i);
    rep(i, N, 2*N) v.at(i) = A.at(i-N);
    
    queue<int> q;
    ll cnt = 0;
    fore(r, v){
        q.push(r);
        cnt += r;
        if (cnt*10 == sum) {
            cout << "Yes" << endl;
            return 0;
        }
        
        while((cnt*10 > sum) || (q.size() > N)){
            cnt -= q.front();
            q.pop();
        }
    }
    cout << "No" << endl;
}